torchvision/io/_video_opt.py

```diff
@@ -495,13 +514,17 @@ def get_pts(time_base):
     return vframes, aframes, _info
 
 
-def _read_video_timestamps(filename, pts_unit="pts"):
+def _read_video_timestamps(
+    filename: str,
+    pts_unit: str = "pts"
+) -> Tuple[Union[List[int], List[Fraction]], Optional[float]]:
     if pts_unit == "pts":
         warnings.warn(
             "The pts_unit 'pts' gives wrong results and will be removed in a "
             + "follow-up version. Please use pts_unit 'sec'."
         )
 
+    pts: Union[List[int], List[Fraction]]
     pts, _, info = _read_video_timestamps_from_file(filename)
```

Comment:
I am bit confused here. You have annotated `_read_video_timestamps_from_file()` above as returning a tuple with its first element being `List[int]`. However, you have annotated `pts` as:
```
pts: Union[List[int], List[Fraction]]
```

Reply:
Yes, because if we don't go in the for loop: it's indeed a `List[int]` as you mentioned
But if you go in the for loop, it becomes a `List[Fraction]` to the best of my understanding :+1: 

If that's not correct, happy to change it!